cfe_internal/enterprise/federation/federation.cf

```diff
@@ -200,17 +208,26 @@ bundle agent transport_user
       handle => "ssh_config_configured",
       edit_line => default:insert_lines("IdentityFile $(ssh_priv_key)");
 
+    selinux_enabled.fix_ssh_context::
+      "$(ssh_home_cil_file)"
+        create => "true",
+        edit_defaults => default:empty,
+        edit_line => default:insert_lines('(filecon "$(home)/.ssh" any (system_u object_r ssh_home_t ((s0)(s0))))');
+
   commands:
-    # Generate a modern ssh keypair (2019-04)
-    "/usr/bin/ssh-keygen"
-      handle => "ssh_keys_configured",
-      args => "-N '' -f $(ssh_priv_key)",
-      if => and( isdir( "$(home)/.ssh" ),
-                 not( fileexists( "$(ssh_priv_key)" )));
-
-    selinux_enabled.missing_ssh_context::
-      "restorecon -R -F  $(home)/.ssh/"
-        contain => default:in_shell;
+    selinux_enabled.fix_ssh_context._stdlib_path_exists_semodule._stdlib_path_exists_restorecon::
+      # the filename becomes the name for the selinux policy (as listed by semodule -l)
+      "$(paths.semodule) -i $(ssh_home_cil_file)";
+      "$(paths.restorecon) -R -F  $(home)/.ssh/";
+
+    any::
+      # Generate a modern ssh keypair (2019-04)
```

Comment:
I'd drop this has comment. Or at least the modern part nothing here indicates anything different from defaults to make it more or less modern.

Reply:
ack